src/tiled-map-resource.ts

```diff
@@ -146,46 +166,48 @@ export class TiledMapResource implements Loadable<TiledMap> {
          }
       }
    }
 
    private _addTiledInsertedTiles(scene: Scene) {
-      const excalibur = this.data?.getExcaliburObjects();
-      if (excalibur.length > 0) {
-         const inserted = excalibur.flatMap(o => o.getInsertedTiles());
-         for (const tile of inserted) {
-            const collisionTypeProp = tile.getProperty<CollisionType>('collisionType');
-            let collisionType = CollisionType.PreventCollision;
-            if (collisionTypeProp) {
-               collisionType = collisionTypeProp.value;
-            }
-            if (tile.gid) {
-               const sprite = this.getSpriteForGid(tile.gid);
-               const actor = new Actor({
-                  x: tile.x,
-                  y: tile.y,
-                  width: tile.width,
-                  height: tile.height,
-                  anchor: vec(0, 1),
-                  rotation: tile.rotation,
-                  collisionType
-               });
-               actor.addComponent(new TiledObjectComponent(tile));
-               if (Flags.isEnabled('use-legacy-drawing')) {
-                  actor.addDrawing(Sprite.toLegacySprite(sprite));
-               } else {
-                  actor.graphics.anchor = vec(0, 1);
-                  actor.graphics.use(sprite);
+      const excaliburObjectLayers = this.data?.getExcaliburObjects();
+      if (excaliburObjectLayers.length > 0) {
+         for (const objectLayer of excaliburObjectLayers) {
+            const inserted = objectLayer.getInsertedTiles();
+            for (const tile of inserted) {
+               const collisionTypeProp = tile.getProperty<CollisionType>('collisionType');
+               let collisionType = CollisionType.PreventCollision;
+               if (collisionTypeProp) {
+                  collisionType = collisionTypeProp.value;
                }
-               scene.add(actor);
-               const z = tile.getProperty<number>('zindex');
-               if (z) {
-                  actor.z = +z.value;
+               if (tile.gid) {
+                  const sprite = this.getSpriteForGid(tile.gid);
+                  const actor = new Actor({
+                     x: tile.x,
+                     y: tile.y,
+                     width: tile.width,
+                     height: tile.height,
+                     anchor: vec(0, 1),
+                     rotation: tile.rotation,
+                     collisionType
+                  });
+                  actor.addComponent(new TiledObjectComponent(tile));
+                  if (Flags.isEnabled('use-legacy-drawing')) {
+                     actor.addDrawing(Sprite.toLegacySprite(sprite));
+                  } else {
+                     actor.graphics.anchor = vec(0, 1);
+                     actor.graphics.use(sprite);
+                  }
+                  scene.add(actor);
+                  const z = tile.getProperty<number>('zindex')?.value ?? (objectLayer.rawObjectGroup.order + this._layerZIndexStart);
```

Comment:
Is `z` supported too? Or is that reserved by Tiled?

Reply:
I can't recall why, it appears to work in latest Tiled when I checked. Originally we went with "zindex" named custom property for some reason (maybe out of fear of it being reserved or colliding?).

I think it's definitely more ergonomic, with the helper suggestion we could easily add support for custom properties named "z" and "zindex"